paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSessionCustomerSheetActivityTest.kt

```diff
@@ -0,0 +1,517 @@
+            isPaymentMethodRemoveEnabled = true,
+            allowsRemovalOfLastSavedPaymentMethod = true,
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+
+            savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "5544").assertIsEnabled()
+
+            val cbcCard = savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            editPage.onRemoveButton().assertIsEnabled()
+        }
+
+    @Test
+    fun `When single CBC card, has remove permissions, and can remove last PM, should be able to remove and edit`() =
+        runTest(
+            cards = listOf(
+                PaymentMethodFactory.card(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = true,
+            allowsRemovalOfLastSavedPaymentMethod = true,
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+
+            val cbcCard = savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            editPage.onRemoveButton().assertIsEnabled()
+        }
+
+    @Test
+    fun `When single CBC card but no remove permissions, can edit but not remove CBC card`() =
+        runTest(
+            cards = listOf(
+                PaymentMethodFactory.card(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = false,
+            allowsRemovalOfLastSavedPaymentMethod = true,
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+
+            val cbcCard = savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            editPage.onRemoveButton().assertDoesNotExist()
+        }
+
+    @Test
+    fun `When single CBC card has remove permissions but cannot remove last, can edit but not remove CBC card`() =
+        runTest(
+            cards = listOf(
+                PaymentMethodFactory.card(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = true,
+            allowsRemovalOfLastSavedPaymentMethod = false,
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+
+            val cbcCard = savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            editPage.onRemoveButton().assertDoesNotExist()
+        }
```

Comment:
All the test above this comment ensure that CBC & remove permissions are properly taken into account when showing whether or not we can remove payment methods in `CustomerSheet`